### PR TITLE
document use cases A/B test and auth header based rate limits

### PR DESCRIPTION
**Pull request by szuecs**
* add auth header based client ratelimits
* add A/B testing section

### CODE REVIEW

**Comment by szuecs** [comment]
:+1:

File: docs/dataclients/kubernetes.md [hunk 1]
```diff
@@ -305,15 +305,35 @@ and [kubernetes dataclient](https://godoc.org/github.com/zalando/skipper/datacli
...
+    spec:
+      rules:
+      - host: app-default.example.org
+        http:
+          paths:
+          - backend:
+              serviceName: app-svc
+              servicePort: 80
+
+If you need to service rate limit service to service communication and
```
Comment (Raffo) [inline docs/dataclients/kubernetes.md]: There is one more "service" here, right?

File: docs/dataclients/kubernetes.md [hunk 2]
```diff
@@ -396,6 +441,90 @@ This ingress route will only allow traffic from networks 1.2.3.0/24 and 195.168.
               servicePort: 80
 
 
+### A/B test
+
+Implementing A/B tests are heavy to build and Skipper can help you to
```
Comment (Raffo) [inline docs/dataclients/kubernetes.md]: Implementing A/B testing is heavy.

File: docs/dataclients/kubernetes.md [hunk 3]
```diff
@@ -396,6 +441,90 @@ This ingress route will only allow traffic from networks 1.2.3.0/24 and 195.168.
...
+do that. You have to have a traffic split somewhere and have your
```
Comment (Raffo) [inline docs/dataclients/kubernetes.md]: " You need to have" sounds better.

**Comment by szuecs** [comment]
:+1:

**Comment by Raffo** [comment]
👍

**Pull request merged**